File: storage/storage-mysql/src/main/java/io/say/better/storage/mysql/domain/entity/Educator.java
package io.say.better.storage.mysql.domain.entity;

import io.say.better.core.common.Provider;
import io.say.better.core.common.RoleType;
import io.say.better.core.common.Status;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Getter
@Entity(name = "EDUCATOR")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Educator extends Member {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "educator_id")
	private Long educatorId;

	@Builder
	private Educator(
			final String email,
			final String birthDate,
			final Provider provider,
			final String providerId,
			final String loginId,
			final String name
	) {
		this.status = Status.ACTIVE;
		this.email = email;
		this.birthDate = birthDate;
		this.role = RoleType.EDUCATOR;
		this.provider = provider;
		this.providerId = providerId;
		this.loginId = loginId;
		this.name = name;
	}

	public static Educator createEducator(
			final String email,
			final String birthDate,
			final Provider provider,
			final String providerId,
			final String loginId,
			final String name
	) {
		return new Educator(email, birthDate, provider, providerId, loginId, name);
	}
}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/domain/entity/Educator.kt
package io.say.better.storage.mysql.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity(name = "EDUCATOR")
class Educator(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "educator_id")
    val educatorId: Long = 0,
    @ManyToOne(targetEntity = Member::class)
    @JoinColumn(name = "member_id")
    var memberId: Member,
    @Column(name = "name", nullable = false, length = 100)
    val name: String = "",
    @Column(name = "birth_date")
    val birthDate: String = "",
    @Column(name = "img_url")
    val imgUrl: String = "",
) : BaseTimeEntity() {
    companion object {
        fun createEducator(
            memberId: Member,
            name: String = "",
            birthDate: String = "",
            imgUrl: String = "",
        ): Educator =
            Educator(
                memberId = memberId,
                name = name,
                birthDate = birthDate,
                imgUrl = imgUrl,
            )
    }
}
